adagrams/game.py

```diff
@@ -1,11 +1,71 @@
+import random
+
+
 def draw_letters():
-    pass
+    letters = (["A"] * 9 + ["B"] * 2 + ["C"] * 2 + ["D"] * 4 + ["E"] * 12
+               + ["F"] * 2 + ["G"] * 3 + ["H"] * 2 + ["I"] * 9 + ["J"]
+               + ["K"] + ["L"] * 4 + ["M"] * 2 + ["N"] * 6 + ["O"] * 8
+               + ["P"] * 2 + ["Q"] + ["R"] * 6 + ["S"] * 4 + ["T"] * 6
+               + ["U"] * 4 + ["V"] * 2 + ["W"] * 2 + ["X"] + ["Y"] * 2
+               + ["Z"])
+
+    hand = random.sample(letters, k=10)
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    for char in word:
+        if word.count(char) > letter_bank.count(char):
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    letter_points = {
+        "A": 1, "E": 1, "I": 1, "O": 1, "U": 1,
+        "L": 1, "N": 1, "R": 1, "S": 1, "T": 1,
+        "D": 2, "G": 2, "B": 3, "C": 3, "M": 3,
+        "P": 3, "F": 4, "H": 4, "V": 4, "W": 4,
+        "Y": 4, "K": 5, "J": 8, "X": 8, "Q": 10,
+        "Z": 10
+    }
+
+    score = 8 if 7 <= len(word) <= 10 else 0
+
+    for char in word:
+        score += letter_points[char.upper()]
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    matching_scores = []
+    word_dict = {}
+  
+    for word in word_list:
+        score = score_word(word)
+        word_dict[word] = score
+
+    max_score = max(word_dict.values())
+
+    # Look for keys with the same score as max_score
+    # Append them to the matching_scores list
+    for word in word_dict:
+        if word_dict[word] == max_score:
+            matching_scores.append(word)
+
+    word_found = False
+    if len(matching_scores) == 1:
+        winning_word = matching_scores[0]
+        word_found = True
+
+    if not word_found:
+        for word in matching_scores:
+            if len(word) == 10:
+                winning_word = word
+                break
+            # Gets smallest word OR first word if all lengths are equal
+            winning_word = min(matching_scores, key=len)
```

Comment:
Let's reorganize this a bit! We've got our `matching_scores`, so let's start with the for loop first, then add the conditional statements within. Whether the length of `matching_scores` is more then 1 or not, we have to do the same thing by assigning our winning word to `winning_word` and `max_score`. Something like:
```python
for word in matching_scores:
    # check the length if it's 10 letters long, return word if so
    # otherwise compare each word's length to the current highest scored word to see if it's shorter
    # reassign current highest scored word if shorter
```